src/net/socket.rs

```diff
@@ -91,11 +92,21 @@ impl Socket {
     }
 
     /// Entry point to the run loop. This should run in a spawned thread since calls to `poll.poll`
-    /// are blocking.
+    /// are blocking. We will default this to sleeping for 1ms.
     pub fn start_polling(&mut self) {
+        self.start_polling_with_duration(Some(Duration::from_millis(1)))
```

Comment:
hardcoded value, alert, we might want to move that into the config. I know the user is able to call  `start_polling_with_duration`,  so I don't have a problem with it.

Reply:
I really wish I could do this in rust... (sigh)
```
pub fn start_polling(&mut self, poll_duration: Option<Duration> = Some(Duration::from_millis(1))) {
```